templates/crossplane/apis/crd.go.tpl

```diff
@@ -21,15 +21,29 @@ type {{ .CRD.Kind }}Parameters struct {
 	// Region is which region the {{ .CRD.Kind }} will be created.
 	// +kubebuilder:validation:Required
 	Region string `json:"region"`
-	{{- range $fieldName, $field := .CRD.SpecFields }}
-	{{- if $field.ShapeRef }}
+{{- range $fieldName, $field := .CRD.SpecFields }}
+	{{ if $field.ShapeRef }}
 	{{ $field.ShapeRef.Documentation }}
 	{{- end }}
-	{{ if $field.IsRequired }} // +kubebuilder:validation:Required
+	{{- if $field.ReferencedType}}
+	{{ $field.Names.Camel }} {{ $field.GoType }} `json:"{{ $field.Names.CamelLower }},omitempty"`
+
+	// {{ $field.Names.Camel }}Ref is a reference to an {{ $field.ReferencedType }} used
+	// to set the {{ $field.Names.Camel }} field.
+	// +optional
+	{{ $field.Names.Camel }}Ref {{if Contains $field.GoType "[]" -}}[]xpv1.Reference{{- else -}}*xpv1.Reference{{- end -}} `json:"{{ $field.Names.CamelLower }}Ref,omitempty"`
+
+	// {{ $field.Names.Camel }}Selector selects references to {{ $field.ReferencedType }}
+	// used to set the {{ $field.Names.Camel }}.
+  // +optional
+	{{ $field.Names.Camel }}Selector *xpv1.Selector `json:"{{ $field.Names.CamelLower }}Selector,omitempty"`
+	{{- else if $field.IsRequired }}
+	// +kubebuilder:validation:Required
 	{{ $field.Names.Camel }} {{ $field.GoType }} `json:"{{ $field.Names.CamelLower }}"`
-	{{- else }} {{ $field.Names.Camel }} {{ $field.GoType }} `json:"{{ $field.Names.CamelLower }},omitempty"` {{ end }}
+	{{- else }}
+	{{ $field.Names.Camel }} {{ $field.GoType }} `json:"{{ $field.Names.CamelLower }},omitempty"`
+	{{- end }}
 {{- end }}
-	Custom{{ .CRD.Kind }}Parameters `json:",inline"`
```

Comment:
did you mean to remove the CustomXXXParameters field inheritance?

Reply:
Yes, we needed only for the additional referencer fields and now that we generate them, it's not necessary anymore.

Reply:
Actually I think we still might need it. Some calls are not recognized by ACK, like additional calls of a resource besides the main ones. I'll add it back.